pkg/ottl/context_inferrer.go

```diff
@@ -0,0 +1,85 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottl // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+
+import "math"
+
+var (
+	defaultContextInferPriority = []string{
+		"log",
+		"metric",
+		"datapoint",
+		"spanevent",
+		"span",
+		"resource",
+		"scope",
+		"instrumentation_scope",
+	}
+)
+
+// ContextInferrer is an interface used to infer the OTTL context from statements paths.
+type ContextInferrer interface {
```

Comment:
Looking at the draft PR I don't believe this needs to be exported, which would be great.

Reply:
Yes, it doesn't necessarily needs to be public, my initial idea was exposing it as a `ottl.ParserCollection` [option](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35050/files#diff-7efe3affb096a0fa16edab4c8e48d665bd120c938103b778ad778b15c5f04ce6R156), so it could support different implementations. 

I'm still have mixing feelings about that, considering we expose the `ottl.Parser[K]`, and that the new `ottl.ParserCollection` (draft) wouldn't be coupled to the existing OTTL contexts, would having a custom context inferrer be useful for something? WDYT?

Anyway, I think you're right, it's probably better not exporting it now and open it in the future if needed.